src/commands/split_column.rs

```diff
@@ -1,31 +1,49 @@
+use crate::commands::WholeStreamCommand;
 use crate::errors::ShellError;
 use crate::object::{Primitive, TaggedDictBuilder, Value};
 use crate::prelude::*;
 use log::trace;
 
-pub fn split_column(
-    args: CommandArgs,
-    registry: &CommandRegistry,
-) -> Result<OutputStream, ShellError> {
-    let args = args.evaluate_once(registry)?;
-    let span = args.name_span();
-    let (input, args) = args.parts();
+#[derive(Deserialize)]
+struct SplitColumnArgs {
+    rest: Vec<Tagged<String>>,
+}
 
-    let positional: Vec<_> = args.positional.iter().flatten().cloned().collect();
+pub struct SplitColumn;
 
-    if positional.len() == 0 {
-        return Err(ShellError::labeled_error(
-            "Split-column needs more information",
-            "needs parameter (eg split-column \",\")",
-            span,
-        ));
+impl WholeStreamCommand for SplitColumn {
+    fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        args.process(registry, split_column)?.run()
     }
 
+    fn name(&self) -> &str {
+        "split-column"
+    }
+
+    fn signature(&self) -> Signature {
+        // TODO: Improve error. Old error had extra info:
+        //
+        //   needs parameter (e.g. split-column ",")
+        Signature::build("split-column").rest()
+    }
+}
+
+fn split_column(
+    SplitColumnArgs { rest: positional }: SplitColumnArgs,
+    RunnableContext { input, name, .. }: RunnableContext,
+) -> Result<OutputStream, ShellError> {
     Ok(input
         .values
         .map(move |v| match v.item {
             Value::Primitive(Primitive::String(ref s)) => {
-                let splitter = positional[0].as_string().unwrap().replace("\\n", "\n");
+                let positional: Vec<_> = positional.iter().map(|f| f.item.clone()).collect();
+
+                // TODO: Require at least 1 positional argument.
+                let splitter = positional[0].replace("\\n", "\n");
```

Comment:
We probably shouldn't merge until this is fixed as it causes a panic.

Reply:
Good catch. I was wondering if it would be possible to add required field before the .rest() part, and then not use positional[_] like we're doing